packages/artillery-plugin-memory-inspector/test/util.mjs

```diff
@@ -0,0 +1,22 @@
+import portfinder from 'portfinder';
+import { spawn } from 'child_process';
+
+export const startTestServer = async () => {
+  const port = await portfinder.getPortPromise({
```

Comment:
The problem is I need to know the port number ahead of time, because I need to start the test with `TEST_PORT`, so that the test can connect to the right port. That's why I didn't rely on a random port assignment.

Reply:
But you spawn the server first, await that, then spawn the test. Am I reading this right?

If you do those in sequence, you know the port ahead of time because its in the result of the spawn server step. Unless you spawn the server and the test concurrently but I see little value in that.

Reply:
Ah the problem is that I don't start the server within the javascript process, but a separate process. I did it that way because I wanted it to be on a separate `pid` to the NodeJS one, as the plugin is trying to test a separate PID.

Normally in a test framework I'd call `server.listen()` from the javascript code directly, in which case I'd be able to do `server.port` like you're saying. But in this case `server` is on a separate file, and I'm using `child_process` to execute that node process, not importing the server object directly.

Reply:
Nice idea @kettanaito 👍 , hadn't thought of that. Will take it into account. There's a few ways test servers are started in Artillery tests (and it's a fairly common thing needed), so we can standardise this in the future.